lib/dao.js

```diff
@@ -624,7 +622,208 @@ DataAccessObject.upsert = function(data, options, cb) {
   }
   return cb.promise;
 };
+/**
+ * Performs upsert based on the instance that matches the where clause
+ . This is a PUT operation and based on
+ * non-primary key. The filter takes one key value pair and updates single matching instance.*
```

Comment:
Still I don't see why `upsertWithWhere` method should not work with PK! It should still work with PK, although it would be the same functionality as `upsert` in this case.